src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -44,6 +46,23 @@ using LogEvents = std::vector<LogEventWithFilterData<LogEvent>>;
  */
 using FilteredLogEventsMap = std::optional<std::vector<size_t>>;
 
+template <typename LogEvent>
+concept GetLogEventIdxInterface = requires(
+        LogEventWithFilterData<LogEvent> const& event,
+        clp::ir::epoch_time_ms_t timestamp
+) {
+    {
+        event.get_timestamp()
+    } -> std::convertible_to<clp::ir::epoch_time_ms_t>;
+};
+
+template <typename LogEvent, typename ToStringFunc>
+concept DecodeRangeInterface = requires(ToStringFunc func, LogEvent const& log_event) {
+    {
+        func(log_event)
+    } -> std::convertible_to<std::string>;
+};
+
```

Comment:
I appreciate the effort to clean this up as well but since this is not directly related to the feature, shall we revert and address it in another PR?